### DISCUSSION TITLE
How VDRplayer/OpenCPN  Determine "Own Ship"

### DISCUSSION THREAD

**Discussion by AlanH70** [transmitterdan/VDRplayer#7] [Q&A]
The sample NMEA .txt file Hakefjord.txt contains numerous types of NMEA 0183 sentences.  GPS data as GLL, RMC, and many others and AIS data as AIVDM and AIVDO sentences.  Upon playback of Hakefjord.txt OpenCPN displays and plots the Own Ship icon and its changing positions and also plots AIS Target icons and their changing positions as separate vessels.  

When the ShipDriver plugin is used to simulate navigation of our vessel it also displays in OpenCPN the Own Ship icon and plots its positions but currently it  records only !AIVDM data for later replay.  When the VDR plugin or the VDRplayer python script replay the ShipDriver recorded track file of !AIVDM sentences OpenCPN interprets the data, of course. as belonging to some AIS Target vessel, not Own Ship.  So OpenCPN displays and position plots the AIS Target icon even though the original context is a ShipDriver record of Own Ship.

One downside of this is the OpenCPN toolbar function "Enable Tracking" works fine if the vessel is interpreted as Own Ship, but not if the vessel is an AIS Target.  The other downside is aesthetic.  A record of own ship navigation should display the Own Ship icon on replay.  

I think this issue probably applies to the ShipDriver plugin, not directly to this VDRplayer script.  But in order to take the issue up with ShipDriver I think we need to know which NMEA sentence[s] transmitted by VDRplayer are interpreted by OpenCPN as Own Ship as is evident somehow is happening in replaying Hakefjord.txt.

**Comment by transmitterdan** [comment]
Have you tried to record the session with VDR plug-in? It will record all messages. Not just AI messages.

An AI message is never used for moving ownship in OpenCPN. It will only use valid GPS messages, not AIS messages. GPS messages usually begin with GP.

I agree that this is not a VDRplayer or OpenCPn issue. You may want to connect with the shipdriver plug-in author on GitHub.
